targets/csharp/source/PlayFabSDK/source/Uunit/tests/PlayFabServerApiTest.cs

```diff
@@ -237,6 +240,60 @@ public void ParallelRequest(UUnitTestContext testContext)
                 }
             });
         }
+
+        /// <summary>
+        /// Multiplayer API
+        /// Try to deliberately request a mutliplayer alias that doesn't exist 
+        ///   Verify that response is MutliplayerServerNotFound, not just Not Found
+        /// </summary>
+        [UUnitTest]
+        public async void TestForNotFoundWithImportantInfo(UUnitTestContext testContext)
+        {
+            PlayFabSettings.staticSettings.TitleId = testTitleData.titleId;
+            PlayFabSettings.staticSettings.DeveloperSecretKey = testTitleData.developerSecretKey;
+
+            var eReq = new AuthenticationModels.GetEntityTokenRequest();
+            eReq.Entity = new AuthenticationModels.EntityKey();
+            eReq.Entity.Type = "title";
+            eReq.Entity.Id = testTitleData.titleId;
+
+            var tokenTask = await PlayFabAuthenticationAPI.GetEntityTokenAsync(eReq);
+
+            var staticPlayer = PlayFabSettings.staticPlayer;
+
+            if(tokenTask.Error != null)
+            {
+                testContext.Fail("Failed to retrieve the Title Entity Token, check your playFabSettings.staticPlayer, are they still logged in? (hint no server api should be called through a logged in client)");
+            }
+
+            if(aliasId == "")
+            {
+                testContext.Fail("aliasId was blank, we will not get the expected failed NotFound response this test is asking for. Make sure testTitleData.json has a valid aliasId listed (check playfab multiplayer dashboard for your own valid aliasId)");
+            }
+
+            MultiplayerModels.UpdateBuildAliasRequest updateBuildAliasRequest = new MultiplayerModels.UpdateBuildAliasRequest()
+            {
+                AliasId = aliasId,
+                AliasName = "aliasName",
+                AuthenticationContext = new PlayFab.PlayFabAuthenticationContext()
+                {
+                    EntityToken = tokenTask.Result.EntityToken
+                },
+            };
+
+            PlayFab.PlayFabResult<MultiplayerModels.BuildAliasDetailsResponse> res = await PlayFab.PlayFabMultiplayerAPI.UpdateBuildAliasAsync(updateBuildAliasRequest);
+
+            string response = PlayFab.PluginManager.GetPlugin<ISerializerPlugin>(PluginContract.PlayFab_Serializer).SerializeObject(res);
+
+            if (response.Contains("MultiplayerServerNotFound") && res.Error.HttpCode == 404)
+            {
+                testContext.EndTest(UUnitFinishState.PASSED, "Detected the Expected MultiplayerServerNotFound PlayFabError");
+            }
+            else
+            {
+                testContext.Fail("We called the Mutliplayer API expecting to not find anything, but we didn't detect this to be the error.");
```

Comment:
You should add response, and the http code to this error output, so we have a better idea what happened in this test from the log.
If there's a 1% failure in Jenkins, it will be a beast to debug. This at least starts us off.